docs/backend/indexing.md

```diff
@@ -0,0 +1,212 @@
+---
+
+(backend-indexing-label)=
+
+# Indexing
+
+To make Plone content searchable, one can use different indexes to index content.
+
+The most important index types are:
+
+- `FieldIndex`
+- `KeywordIndex`
+- `DateIndex`
+- `DateRangeindex`
+- `BooleanIndex`
+- `ZCTextIndex`
+
+The most important indexes are:
+
+## `SearchableText`
+
+The `SearchableText` is a `ZCTextIndex` for indexing full text and is used by default for `Dublin core` fields like Title, Description and Text.
+
+### TextIndexer
+
+To add other fields to the `SearchableText`, one can use the `plone.app.dexterity.textindexer`.
+
+For enabling the indexer just add the behavior to the list of behaviors of your content types.
+
+In your *profiles/default/types/YOURTYPE.xml* add the behavior::
+
+    <?xml version="1.0"?>
+    <object name="example.conference.presenter" meta_type="Dexterity FTI"
+       xmlns:i18n="http://xml.zope.org/namespaces/i18n"
+       i18n:domain="example.conference">
+
+     <!-- enabled behaviors -->
+     <property name="behaviors" purge="false">
+         <element value="plone.textindexer" />
+     </property>
+
+    </object>
+
+
+Now you need to mark the fields you want to have in your SearchableText. 
+This  can be done with the searchable directive:
+
+```python
+from plone.app.dexterity import textindexer
+from plone.supermodel.model import Schema
+from plone import schema
+
+class IMyBehavior(Schema):
+
+    textindexer.searchable('specialfield')
+    specialfield = schema.Text(title=u'Special field')
+
+```
+
+If you want to mark fields of an existing 3rd party behavior, it can be done using this utility function:
+
+```python
+from plone.app.dexterity.behaviors.metadata import ICategorization
+from plone.app.dexterity.textindexer import utils
+
+utils.searchable(ICategorization, 'categorization')
+```
+
+The `title` and `description` on `plone.app.dexterity`'s `IBasic` behavior are marked as searchable by default.
+For marking them as no longer searchable, there is a utility function:
+
+```python
+from plone.app.dexterity.behaviors.metadata import IBasic
+from plone.app.dexterity.textindexer import utils
+
+utils.no_longer_searchable(IBasic, 'title')
+```
+
+Alternatively, if you specified your model as a plone.supermodel XML model,
+you can mark the field searchable by using `indexer:searchable="true"`:
+
+```xml
+
+<model xmlns="http://namespaces.plone.org/supermodel/schema"
+        xmlns:indexer="http://namespaces.plone.org/supermodel/indexer">
+    <schema based-on="plone.supermodel.model.Schema">
+
+        <field name="specialfield" type="zope.schema.TextLine"
+                indexer:searchable="true">
+        <title>Special field</title>
+        </field>
+
+    </schema>
+</model>
+```
+
+
+Your SearchableText indexer includes now your custom field on your behavior. 
+It is enabled on your content type, if the `plone.textindexer` behavior is enabled there too.
```

Comment:
The SearchableText indexer includes now your custom field of your behavior. 
The field of your content type is indexed if the `plone.textindexer` behavior is enabled on your content type.